### PR TITLE
Schedule Component And Mockup Backend Server

### PR DESCRIPTION
**Pull request by stephypy**
**Note: The following PR is closed due to the successful implementation of the Google Calendar API in browser**

This PR includes a mock implementation of making calls to the backend server for the schedule page. 

* It currently only displays one task at a time
* Three random tasks were hardcoded and randomly selected for display
* The datepicker does not affect the displayed tasks yet; this will be done in a separate PR after both this PR and PR#16 get approved and merged
* Different tasks get displayed every time the page is refreshed; the screenshot below shows an example of one of the hardcoded tasks for mock implementation

image

Future updates: 
This will serve as a basis for the calendar API. The calendar API will be helpful in the following ways:
* Create a calendar when a garden is created
* Calendar is shared with the members of the garden
* Admin can create 'events' (tasks) that can be sent to different members in the garden
* The calendar API allows getting events for a particular day; each event includes attributes such as title, time, participants, and description

Potential complications:
* Duplicate calendar names

### CODE REVIEW

File: src/main/java/com/google/growpod/data/Task.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// limitations under the License.
+
+package com.google.growpod.data;
+
+import java.util.List;
+
+/** Task data class * */
+public class Task {
+
+  /* The title of the event */
```
Comment (codyjrivera) [inline src/main/java/com/google/growpod/data/Task.java:22]: You should convert these into JavaDocs by changing these comments to, for example `/** The title of the event. */`, (ensuring you have a period at the end of the first sentence to pass the linter).

File: src/main/java/com/google/growpod/data/Task.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  public Task(String title, String time, List<String> participants, String description) {
+    this.title = title;
+    this.time = time;
+    this.participants = participants;
+    this.description = description;
+  }
+
+  /* Getters */
+  public String getTitle() {
```
Comment (codyjrivera) [inline src/main/java/com/google/growpod/data/Task.java:42]: Nit: The rest of the files clump getters and setters together by field. For example, `getTitle` would be followed by `setTitle`.

File: src/main/java/com/google/growpod/servlets/TaskServlet.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that handles the tasks from a particular garden and date
+ *
+ * TODO: This is the mockup implementation; a future PR will implement calendar api and connect gardens together
+ */
+
+@WebServlet({"/schedule", "schedule/*"})
```
Comment (codyjrivera) [inline src/main/java/com/google/growpod/servlets/TaskServlet.java:33]: Change `"schedule/*"` to `"/schedule/*"`, to be consistent with other servlets.

File: src/main/java/com/google/growpod/data/Task.java (hunk 2 above)
Comment (stephypy) [reply src/main/java/com/google/growpod/data/Task.java:42]: Thank you for the comment!

File: src/main/java/com/google/growpod/servlets/TaskServlet.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+@WebServlet({"/schedule", "/schedule/*"})
+public class TaskServlet extends HttpServlet {
+
+  private List<Task> tasks;
+
+  /* Mockup implementation: Hardcoded tasks for testing */
+  @Override
+  public void init() {
+    tasks = new ArrayList<>();
+    List<String> hosts = new ArrayList();
```
Comment (caroqliu) [inline src/main/java/com/google/growpod/servlets/TaskServlet.java:42]: This is fine for mock data, but what would `hosts` and `interns` actually be with real data? `admin` and `volunteers`? Why is the differentiation needed?

File: src/main/webapp/src/app/model/task-info.model.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * TaskInfo interface
+ *
+ */
+export interface TaskInfo {
+  title: string;
+  time: string;
+  participants: string;
+  description: string;
```
Comment (caroqliu) [inline src/main/webapp/src/app/model/task-info.model.ts:23]: I think this is optional?

File: src/main/webapp/src/app/task/task.component.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnInit} from '@angular/core';
+import {HttpClient, HttpResponse} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {TaskInfo} from '../model/task-info.model';
+
+@Component({
+  selector: 'task',
```
Comment (caroqliu) [inline src/main/webapp/src/app/task/task.component.ts:21]: nit: Maybe this should be `task-list`? Is it possible to display more than one task in one instance of this component?

**Pull request closed without merging**